File: advent-of-code-2021/day-01.rb
INPUT = File.read("day-01.input").split("\n").map(&:to_i)

# INPUT = %w{199
# 200
# 208
# 210
# 200
# 207
# 240
# 269
# 260
# 263}.map(&:to_i)

increases = INPUT.each_cons(2).select { |(a, b)| b > a }.count

puts "Part One: %d" % increases

windows = INPUT.each_cons(3).map { |window| window.sum }.each_cons(2).select { |(a, b)| b > a }.count

puts "Part Two: %d" % windows

File: advent-of-code-2021/day_01.erl
-module(day_01).
-export([main/1]).
-mode(compile).


count_incr([A | [B | _] = Rest], Acc) when B > A ->
	count_incr(Rest, Acc + 1);
count_incr([_ | Rest], Acc) ->
	count_incr(Rest, Acc);
count_incr([], Acc) ->
	Acc.

count_incr(Rest) ->
	count_incr(Rest, 0).


count_incr3([A | [_, _, B | _] = Rest], Acc) when B > A ->
	count_incr3(Rest, Acc + 1);
count_incr3([_ | Rest], Acc) ->
	count_incr3(Rest, Acc);
count_incr3([], Acc) ->
	Acc.

count_incr3(Rest) ->
	count_incr3(Rest, 0).


main([File]) ->
	{ok, Input} = ekk:read_lines(File, fun binary_to_integer/1),
	Count = count_incr(Input),
	io:format("Part One: ~w\n", [Count]),
	Count2 = count_incr3(Input),
	io:format("Part Two: ~w\n", [Count2]).
